headeredevent.go

```diff
@@ -24,23 +24,22 @@ type EventHeader struct {
 // when marshalling into JSON and will be separated out when unmarshalling.
 type HeaderedEvent struct {
 	EventHeader
-	Event
+	*Event
 }
 
 // Unwrap extracts the event object from the headered event.
-func (e *HeaderedEvent) Unwrap() Event {
+func (e *HeaderedEvent) Unwrap() *Event {
 	if e.RoomVersion == "" {
 		// TODO: Perhaps return an error here instead of panicing
 		panic("gomatrixserverlib: malformed HeaderedEvent doesn't contain room version")
 	}
-	event := e.Event
-	event.roomVersion = e.RoomVersion
-	return event
```

Comment:
Why did you change this?

Reply:
Because it's no longer taken by value, therefore we can just make sure that the actual event's inner room version is right and return a reference to that directly.